pkg/ccl/changefeedccl/schema_registry.go

```diff
@@ -228,7 +225,7 @@ func (r *confluentSchemaRegistry) doWithRetry(ctx context.Context, fn func() err
 		}
 		log.VInfof(ctx, 2, "retrying schema registry operation: %s", err.Error())
 	}
-	return changefeedbase.MarkRetryableError(err)
+	return changefeedbase.WithTerminalError(err)
```

Comment:
I'm not sure we should be doing that -- terminal error stuff.  We retry all errors that come up due to downstream failure..  Failure to contact external schema registry should be treated same way.  Retry few times, then fail with retryable error.

Reply:
What we should do instead is a) change retry to be something more reasonable than 2.  Make it 10; make it backoff more aggressively.  Better yet, use `retry.go`.

Also, what we should do is to add a metric for schema registry.
I know it's more of an addition, but I think the lack of any observability was detrimental here.

Reply:
I'm ok with using retry.go here and adding observability, but I think this should be a permanent error. Changefeeds will retry forever if they see retryable errors right? If this code retries 10 times and fails, the changefeed should probably fail. For example, what if they get a 401 due to incorrect SSL parameters?

Retrying forever is a customer concern: https://cockroachlabs.slack.com/archives/C9TGBJB44/p1678197741759369?thread_ts=1677783523.194569&cid=C9TGBJB44

Reply:
> I'm ok with using retry.go here and adding observability, but I think this should be a permanent error. Changefeeds will retry forever if they see retryable errors right? If this code retries 10 times and fails, the changefeed should probably fail. For example, what if they get a 401 due to incorrect SSL parameters?
> 
> Retrying forever is a customer concern: https://cockroachlabs.slack.com/archives/C9TGBJB44/p1678197741759369?thread_ts=1677783523.194569&cid=C9TGBJB44

401 may very well happen, but why is schema registry should be treated differently from say kafka:// URI? or webhook?
We retry forever in those cases.  I think consistency here is better.
Furthermore, customer concerns around retries forever could be better resolved 1. by adding better tools to validate sink *and* schema registry validation (btw... do we validate schema registry when we create changefeed? -- that would be a good place to fail).  2. Customers should monitor stats, including retry rates, and alert on those.